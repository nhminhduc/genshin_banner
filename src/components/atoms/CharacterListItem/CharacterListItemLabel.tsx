import cx from "classnames";
import { startCase } from "lodash";

interface CharacterListItemLabelProps {
  name: string;
  noVersions?: number;
}

const CharacterListItemLabel = ({
  name,
  noVersions,
}: CharacterListItemLabelProps) => (
  <div
    className={cx(
      "flex bg-[#e4c179] left-0 right-0 bottom-0 justify-between items-center h-8 px-1 py-2",
    )}
  >
    <div className={cx("font-bold text-sm text-slate-600")}>
      {startCase(name)}
    </div>
    {(noVersions || noVersions === 0) && (
      <div
        className={cx("font-bold text-xl", {
          "text-green-600": noVersions === 0,
          "text-lime-300": noVersions <= 3,
          "text-rose-600": noVersions > 3 && noVersions < 10,
          "text-rose-900": noVersions >= 10,
        })}
      >
        {noVersions}
      </div>
    )}
  </div>
);

export default CharacterListItemLabel;
