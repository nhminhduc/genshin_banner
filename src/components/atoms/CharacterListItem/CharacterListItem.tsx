import { ImageListItem } from "@mui/material";
import { elements } from "assets/images";
import cx from "classnames";

import CharacterListItemLabel from "./CharacterListItemLabel";

interface CharacterListItemProps {
  name: string;
  visionKey: string;
  noVersions?: string;
  found?: boolean;
}

const CharacterListItem = ({
  name,
  visionKey,
  noVersions = "0",
  found,
}: CharacterListItemProps) => {
  return (
    <ImageListItem
      className={cx(
        "bg-amber-100/50 rounded border-2 border-[#ebe7df] border-solid flex",
        { "opacity-30": !found }
      )}
    >
      <span>
        <img alt={visionKey} className="w-8 h-8 absolute right-0" src={elements[visionKey as keyof typeof elements]} />
      </span>
      <img
        alt={name}
        className={cx("characterImage", "w-full h-full truncate")}
        loading="lazy"
        src={`characters/${name}/icon-big.webp`}
      />
      <CharacterListItemLabel name={name} noVersions={noVersions} />
    </ImageListItem>
  )
};

export default CharacterListItem;
