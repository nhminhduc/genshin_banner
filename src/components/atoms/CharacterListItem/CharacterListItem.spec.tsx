import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import CharacterListItem from "./CharacterListItem";

describe("CharacterListItem test", () => {
  test("CharacterListItem snapshot", () => {
    const { container } = render(
      <CharacterListItem
        link="https://google.com"
        name="test CharacterListItem"
        rarity="4"
        visionKey="pyro"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
