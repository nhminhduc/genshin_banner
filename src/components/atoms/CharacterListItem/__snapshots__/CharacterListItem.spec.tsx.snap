// Vitest Snapshot v1

exports[`CharacterListItem test > CharacterListItem snapshot 1`] = `
<div>
  <li
    class="MuiImageListItem-root rounded border-2 border-[#ebe7df] border-solid flex group hover:scale-[1.2] hover:z-10 hover:transition-all hover:cursor-pointer opacity-30 bg-[#9470bb] css-l086p4-MuiImageListItem-root"
    style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
  >
    <span>
      <img
        alt="pyro"
        class="w-8 h-8 absolute right-0"
        src="elements/pyro/icon.png"
      />
    </span>
    <img
      alt="test CharacterListItem"
      class="MuiImageListItem-img characterImage w-full h-full truncate"
      loading="lazy"
      src="characters/test CharacterListItem/icon-big.png"
    />
    <div
      class="flex bg-[#e4c179] left-0 right-0 bottom-0 justify-between items-center h-8 px-1 py-2"
    >
      <div
        class="font-bold text-sm text-slate-600 group-hover:text-rose-600"
      >
        Test Character List Item
      </div>
    </div>
  </li>
</div>
`;
